### ISSUE TITLE
AOT compilation fails with strictNullChecks when angular2-toaster included

### ISSUE THREAD

**Issue by M1strall** [Stabzs/Angular2-Toaster#111]
Version 3.0.1.
We were using angular2-toaster and recently migrated to Angular4. Now we are trying to enable AOT compilation.
Toaster works fine for Dev and regular builds but 
I'm getting following error with AOT build (strict null checks enabled):

```TypeScript
Starting compilation using the angular compiler.
Angular compilation done, starting webpack bundling.
Error: Error at /Users/user/projects/project/compiled/node_modules/angular2-toaster/src/toast.component.ngfactory.ts:223:27: Object is possibly 'undefined'.
Error at /Users/user/projects/project/compiled/node_modules/angular2-toaster/src/toast.component.ngfactory.ts:225:27: Object is possibly 'undefined'.
    at UserError.Error (native)
    at new UserError (/Users/user/projects/tools/@angular/tsc-wrapped/src/tsc.ts:33:5)
    at check (/Users/user/projects/tools/@angular/tsc-wrapped/src/tsc.ts:81:11)
    at Tsc.typeCheck (/Users/user/projects/tools/@angular/tsc-wrapped/src/tsc.ts:175:5)
    at /Users/user/projects/tools/@angular/tsc-wrapped/src/main.ts:128:11
    at process._tickCallback (internal/process/next_tick.js:103:7)
```

If I open toast.component.ngfactory.ts IntelliJ underlines me following lines:
```TypeScript
    var co:import2.ToastComponent = v.component;
    const currVal_0:any = 'toaster-icon';
    const currVal_1:any = co.iconClass;
    ck(v,3,0,currVal_0,currVal_1);
    const currVal_2:any = co.toast.toasterConfig.titleClass;
    ck(v,8,0,currVal_2);
    const currVal_4:any = co.toast.toasterConfig.messageClass;
```
toasterConfig.titleClass & toasterConfig.messageClass.

My guess they are used in toaster template without nullability check for titleClass & messageClass:
```html
  <i class="toaster-icon\" [ngClass]="iconClass"></i>
<div class="toast-content">
<div [ngClass]="toast.toasterConfig.titleClass">{{toast.title}}</div>
<div [ngClass]="toast.toasterConfig.messageClass" [ngSwitch]="toast.bodyOutputType">
<div *ngSwitchCase="bodyOutputType.Component" #componentBody></div>
<div *ngSwitchCase="bodyOutputType.TrustedHtml" [innerHTML]="toast.body"></div>               <div *ngSwitchCase="bodyOutputType.Default">{{toast.body}}</div></div></div>
<div class="toast-close-button" *ngIf="toast.showCloseButton" (click)="click($event, toast)"            [innerHTML]="safeCloseHtml"></div>"
```

Disabling strictNullChecks solves issue but it forces to disable this option for whole project.

**Comment by Stabzs** [comment]
In looking into this, compiling the project with the `strictNullCheck` flag set (apologies for not having this set already and thank you for bringing it to my attention) does not seem to cause compile-time issues within the template.

I am creating tests around AOT-centric applications consuming the library to track down the root cause.

**Comment by M1strall** [comment]
Regular compilation with strictNullChecks does not check html templates,
AOT creates typesafe factories that lead to additional checking in those

**Comment by pgrm** [comment]
Can't this be solved simply by using ?. in the template? Or would that cause other issues?

**Comment by Stabzs** [comment]
@pgrm I think you hit the nail on the head...this is probably the best solution.

I spent some time working through potentially changing the `ToasterConfig` interface to make those properties non-nullable, but that fundamentally changed the meaning of the interface which isn't at all the right move.

I've made this modification and it works well based on the test suite and the manual tests I've been running.  I'm working to push this into a new release as soon as possible.
